### PR TITLE
Skip running tax calculation tests based on date

### PR DESCRIPTION
**Pull request by polmiro**
## Motivation / Description

Re-enables tax e2e tests and adds a new env to allow deactivating them based on date to circumvent possible rate limit issues.

### CODE REVIEW

File: examples/webbilling-demo/src/tests/helpers/integration-test.ts [hunk 1]
```diff
@@ -5,7 +5,43 @@ import { getEmailFromUserId } from "./test-helpers";
...
+  if (!skipUntilDate) return false;
+  try {
+    // Validate the format is yyyy-mm-dd
+    const dateFormatRegex = /^\d{4}-\d{2}-\d{2}$/;
+    if (!dateFormatRegex.test(skipUntilDate)) {
+      console.error(
+        "Invalid date format for VITE_SKIP_TAXES_TESTS_UNTIL. Expected format is 'yyyy-mm-dd'",
+      );
+      return false;
+    }
```
Comment (nicfix) [inline examples/webbilling-demo/src/tests/helpers/integration-test.ts:19]: Nit: Maybe this check is not needed given that we are passing the string to the Date object and checking if we created a valid object.

But I might not see the edge case here.

**Review by nicfix — APPROVED** [review]

File: examples/webbilling-demo/src/tests/helpers/integration-test.ts (hunk 1 above)
Comment (polmiro) [reply examples/webbilling-demo/src/tests/helpers/integration-test.ts:19]: It accepts a couple of different formats but I decided to make it explicit on "yyyy-mm-dd" to avoid confusion (epoch, iso timestamp, iso date, etc).
